app/controllers/api/v2/user_columns_controller.rb

```diff
@@ -0,0 +1,56 @@
+module Api
+  module V2
+    class UserColumnsController < V2::BaseController
+      include Api::Version2
+      before_action :find_user_column, :only => [:reset]
+
+      api :GET, "/user_columns/", N_("List of user columns for a user")
+      def index
+        @user_columns = User.current.user_columns
+      end
+
+      api :GET, "/user_columns/:resource", N_("List of user columns for a resource")
+      param :resource, String, :required => true
+      def columns
+        render :json => { root_node_name => UserColumn.formatted_columns(params[:resource])}
+      end
+
+      def_param_group :user_column do
+        param :user_columns, Hash, :required => true do
+          param :resource, String, :required => true, :desc => N_("name of the resource")
+          param :columns, Array, :desc => N_("List of user selected columns")
+        end
+      end
+
+      api :POST, "/user_columns/", N_("Creates a user column for a given resource")
+      param_group :user_column
+      def create
+        @user_column = UserColumn.where(:user => User.current.id, :resource => params[:resource]).first_or_create
+        @user_column.columns = params[:columns]
+        process_response @user_column.save
+      end
+
+      api :DELETE, "/user_columns", N_("Delete columns for a resource")
+      param :resource, String, :required => true, :desc => N_("name of the resource")
+      def reset
+        process_response @user_column.destroy_all
```

Comment:
this will always remove all? just like destroy_all?

in general I would prefer to have one destroy method that get an array of ids, so it can delete one, many or all.